src-ts/tools/learn/free-code-camp/FreeCodeCamp.tsx

```diff
@@ -1,45 +1,47 @@
-import { Dispatch, FC, memo, MutableRefObject, SetStateAction, useCallback, useEffect, useMemo, useRef, useState } from 'react'
+import { Dispatch, FC, SetStateAction, useCallback, useContext, useEffect, useMemo, useState } from 'react'
 import { NavigateFunction, useNavigate, useSearchParams } from 'react-router-dom'
 
-import { EnvironmentConfig } from '../../../config'
 import {
     Breadcrumb,
     BreadcrumbItemModel,
     LoadingSpinner,
     Portal,
+    profileContext,
+    ProfileContextData,
 } from '../../../lib'
 import {
     CourseOutline,
     CoursesProviderData,
     LearnLesson,
     LearnModule,
     LessonProviderData,
+    MyCertificationProgressProviderData,
+    startMyCertificationsProgressAsync,
+    updateMyCertificationsProgressAsync,
+    UpdateMyCertificateProgressActions,
     useCoursesProvider,
     useLessonProvider,
+    useMyCertificationProgress,
 } from '../learn-lib'
 import { getFccLessonPath } from '../learn.routes'
 
 import { CollapsiblePane } from './collapsible-pane'
+import { FccFrame } from './fcc-frame'
 import styles from './FreeCodeCamp.module.scss'
 import { TitleNav } from './title-nav'
 
-const FreecodecampIfr: FC<any> = memo((params: any) => (
-    <iframe
-        className={styles.iframe}
-        ref={params.frameRef}
-    />
-))
-
 const FreeCodeCamp: FC<{}> = () => {
+    const { profile }: ProfileContextData = useContext(profileContext)
+
     const navigate: NavigateFunction = useNavigate()
     const [searchParams]: any = useSearchParams()
 
-    const frameRef: MutableRefObject<HTMLElement|any> = useRef()
-    const frameIsReady: MutableRefObject<boolean> = useRef<boolean>(false)
     const [courseParam, setCourseParam]: [string, Dispatch<SetStateAction<string>>] = useState(searchParams.get('course') ?? '')
     const [moduleParam, setModuleParam]: [string, Dispatch<SetStateAction<string>>] = useState(searchParams.get('module') ?? '')
     const [lessonParam, setLessonParam]: [string, Dispatch<SetStateAction<string>>] = useState(searchParams.get('lesson') ?? '')
 
+    const { certificateProgress, setCertificateProgress }: MyCertificationProgressProviderData = useMyCertificationProgress(profile?.userId, courseParam)
```

Comment:
Need to change this to use the `certificationProgressId` to identify the specific progress record.

Reply:
This is the same case as above (CourseDetailsPage). I'm now sending the provider as well.